packages/auth/src/providers/cognito/utils/signInHelpers.ts

```diff
@@ -630,3 +633,19 @@ export function isMFATypeEnabled(
 	if (!mfaTypes) return false;
 	return mfaTypes.includes(mfaType);
 }
+
+export async function isUserAuthenticated() {
+	let authUser: AuthUser | undefined;
+	try {
+		authUser = await getCurrentUser();
+	} catch (error) {}
+
+	if (authUser && authUser.userId && authUser.username) {
+		throw new AuthError({
+			name: USER_ALREADY_AUTHENTICATED_EXCEPTION,
+			message:
+				'There is already a signed in user because auth tokens were found.',
+			recoverySuggestion: 'Please call signOut before calling signIn again.',
```

Comment:
Just to match tone of the other recovery suggestions elsewhere in the lib.

```suggestion
			recoverySuggestion: 'Call signOut before calling signIn again',
```